### ISSUE TITLE
IllegalStateException: Duplicate Key

### ISSUE THREAD

**Issue by Dessie0** [felldo/JEmoji#1]
When using the emojis.json found [here](https://github.com/felldo/JEmoji/blob/master/lib/src/main/resources/emojis.json), there's a duplicate key error when it tries to map the `EMOJI_UNICODE_TO_EMOJI` map. 

This is the full stacktrace:

```
Caused by: java.lang.IllegalStateException: Duplicate key �? (attempted merging values Emoji{emoji='�?', unicode='?', discordAliases=[:grin:], githubAliases=[:grin:], slackAliases=[:grin:], hasFitzpatrick=false, hasHairStyle=false, version=0.6, qualification=FULLY_QUALIFIED, description='beaming face with smiling eyes', group=SMILEYS_AND_EMOTION, subgroup=FACE_SMILING} and Emoji{emoji='�?', unicode='?', discordAliases=[:heart_eyes:], githubAliases=[:heart_eyes:], slackAliases=[:heart_eyes:], hasFitzpatrick=false, hasHairStyle=false, version=0.6, qualification=FULLY_QUALIFIED, description='smiling face with heart-eyes', group=SMILEYS_AND_EMOTION, subgroup=FACE_AFFECTION})
```

When I go into the debugger, it's pretty easy to tell that when the JSON file is read, the `emoji` field for both 😁 and 😍 are the same, as seen in the image below.
 
![image](https://github.com/felldo/JEmoji/assets/42523509/d823de0c-1a85-433a-a77d-05b9e2562952)

 Perhaps mapping with the Unicode instead of Emoji would fix this?

**Comment by felldo** [comment]
Hi, thank you for reporting this potentual issue.

I tried to reproduce this without success. It works just fine for me.
What version and file encoding do you use?

> Perhaps mapping with the Unicode instead of Emoji would fix this?

They unicode and emoji fields are the same. If you would `equals` both Strings, it would return true. It's just visually different like you can see in the emoji JSON.
Regarding your debugger, it may be possible that your console encoding is set to something different than `UTF-8`. To change it go to `Settings` -> `Editor` -> `General` -> `Console`. After doing that, the emoji and unicode should look the same in your debugger.

**Comment by Dessie0** [comment]
My debugger encoding was set to windows-1252, but changing it to UTF-8 doesn't seem to change anything.

I'm using Java 17 and when running my application I am using `-Dfile.encoding=UTF-8` to make sure it's running with UTF-8 encoding. All the project files *should* be in UTF-8 as well. 

I forked and did some testing on my own, changing the `EMOJI_UNICODE_TO_EMOJI` key map function to `Emoji::getUnicode` from `Emoji::getEmoji` atleast allowed it to run for me, though I had to make changes to some methods. I couldn't get all of the tests to pass though, so I haven't made a PR. Might be worth looking into even if they're the same?

**Comment by felldo** [comment]
> All the project files should be in UTF-8 as well.

In IntelliJ you can see the file encoding in the bottom right of your IDE.

Good thing you tried to debug this by yourself by forking the project.
So there needs to be some computer specific differences between our setups. Could you please try to add
```kts
tasks.withType<JavaCompile> {
    options.encoding = "UTF-8"
}

tasks.withType<Test> {
    systemProperty("file.encoding", "UTF-8")
}

tasks.withType<Javadoc>{
    options.encoding = "UTF-8"
}
``` 
to the `build.gradle.kts` file to see if it makes any differences after reloading the configuration?

**Comment by Dessie0** [comment]
Ah, yes. That fixed it. I made a PR to add those. 😄 Thank you!

**Comment by felldo** [comment]
I have released a new version `1.1.5` which includes your fix. It should be available shortly
